### PR TITLE
Convert transfer_submitter role to implicitly-defined

### PR DESCRIPTION
**Pull request by matt-bernhardt**
#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

**Comment by matt-bernhardt** [comment]
CodeClimate is flagging the cognitive complexity of the initialize method in `ability.rb` - but I'm not sure how to avoid this level of complexity (6, one more than the threshold of 5). It is possible that an early return instead of a nested user.present? check would lower the complexity?

**Comment by JPrevost** [comment]
@matt-bernhardt I think the codeclimate thing is fine. It's approaching "let's think about this" level but it reads pretty clean to me. I'll make a note of that in codeclimate as part of the code review I'm doing now.

File: app/models/ability.rb [hunk 1]
```diff
@@ -13,6 +13,10 @@ def initialize(user)
         can :manage, :all
       end
 
+      if (user.submitter? && @user.role == "basic")
```
Comment (JPrevost) [inline app/models/ability.rb:16]: I think a brief comment such as `promote submitters that don't already have higher roles` might help us realize in the future _why_ we are doing this. (non-blocking)

**Review by JPrevost — APPROVED** [review]
Very minor suggestion on a method to document. 👍🏻

**Comment by matt-bernhardt** [comment]
After conversations on Slack, I'm going to go ahead and merge this now. I'll make Stephanie and Mikki into Submitters in the staging app for their review, and we can spawn further tickets with any feedback that they provide.

**Pull request merged**